app/javascript/components/create-nuage-cloud-subnet-form.jsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import NuageCloudSubnetForm from './nuage-cloud-subnet-form'
+
+const API = window.API;
+
+const createSubnet = (values, emsId, routerRef) => {
+  API.post(`/api/providers/${emsId}/cloud_subnets`, {
+    action: 'create',
+    resource: {...values, router_ref: routerRef},
+  }).then(response => {
+    let res = response['results'][0];
+    window.add_flash(res.message, res.success ? 'success' : 'error');
+  })
+};
+
+class CreateNuageCloudSubnetForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleFormStateUpdate = this.handleFormStateUpdate.bind(this);
+    this.routerRef = 'ebca9d3b-53a4-4fd4-8268-196cb3fa6072'; // TODO: pass from Rails
+    this.emsId = 4;                                          // TODO: pass from Rails
```

Comment:
generally you have `ManageIQ.record.recordId` with the id of the entity being displayed. If you need some other params, then the best way is to fetch them via API starting with the recordId.

Passing arguments through buttons from the toolbar will not help. Those are stored in a singleton evaluated at the first use of a toolbar (when the toolbar is `require`d).

Reply:
Thanks, will apply Monday morning. We had a discussion with @Hyperkid123 and came to the same conclusion 👍 . BTW UI team did such a great job supporting React Components that I'm preparing a blogpost of it. Thank you so much for this.